sourcegraph/sourcegraph
# cody: admin page

Adds a Cody admin page that will be used to configure everything Cody related. Currently, it will be used to schedule repo embedding jobs if embeddings are enabled in the site config.



## Screenshots

<img width="948" alt="Screenshot 2023-03-01 at 09 09 32" src="https://user-images.githubusercontent.com/6417322/222081260-8b61f4f9-3219-4c82-ac88-41525f303608.png">

<img width="968" alt="Screenshot 2023-03-01 at 09 09 50" src="https://user-images.githubusercontent.com/6417322/222081256-f5721ac0-4753-4271-94c9-82801ed5e373.png">

## Test plan


* Enable embeddings in the site config
* Go to `/site-admin/cody`
* Find a couple of repositories and schedule them for embedding
* The list of jobs should be automatically refreshed with their state
* Wait a couple of minutes, and refresh the page
* The embedding jobs should hopefully be completed
